src/http/adapter.js

```diff
@@ -68,6 +68,14 @@ function checkPermissionFor( state, user, context, action ) {
 		.then( onPermission, onError );
 }
 
+function executeStack( resource, envelope, stack ) {
+	var calls = stack.slice( 0 );
+	var call = calls.shift();
+	return call ?
+		call.bind( resource )( envelope, executeStack.bind( undefined, resource, envelope, calls ) ) :
+		{ status: 500, data: { message: 'The server failed to provide a response' } };
+}
```

Comment:
Hopefully the README made that clear? You _must_ return something. Either `return next();` or you return a promise or the data structure that defines the response. If you return something other than the next call result, it does short circuit. You can't have both, but then that's pretty standard as far as middleware goes that I'm familiar with - I don't know that I'd want to see us returning stuff AND continuing down the stack.